include/rom.h

```diff
@@ -3,84 +3,109 @@
 
 #include "xlFileGCN.h"
 
+typedef s32 pCallback_func(void);
+typedef s32 pProgressCallback_func(f32);
+
 // __anon_0x4CF87
-// size: 0x4
 typedef enum RomModeLoad {
-    RLM_NONE = -1,
-    RLM_PART = 0,
-    RLM_FULL = 1,
-    RLM_COUNT_ = 2
+    /* -1 */ RLM_NONE = -1,
+    /*  0 */ RLM_PART,
+    /*  1 */ RLM_FULL,
+    /*  2 */ RLM_COUNT
 } RomModeLoad;
 
-// __anon_0x4CFE6
-// size: 0x10
+// __anon_0x5219D
+typedef enum RomCacheType {
+    /* -1 */ RCT_NONE = -1,
+    /*  0 */ RCT_RAM,
+    /*  1 */ RCT_ARAM
+} RomCacheType;
+
+// __anon_0x4CFE6 
 typedef struct RomBlock {
-    int iCache; // 0x0
-    u32 nSize; // 0x4
-    u32 nTickUsed; // 0x8
-    char keep; // 0xC
-} RomBlock;
+    /* 0x00 */ s32 iCache;
+    /* 0x04 */ u32 nSize;
+    /* 0x08 */ u32 nTickUsed;
+    /* 0x0C */ s8 keep;
+} RomBlock; // size = 0x10
 
-// size: 0x14
-typedef struct __anon_0x4D0FA {
-    int bWait; // 0x0
-    int (*pCallback)(); // 0x4
-    u8* pTarget; // 0x8
-    u32 nSize; // 0xC
-    u32 nOffset; // 0x10
-} __anon_0x4D0FA;
+typedef struct RomCopyState {
+    /* 0x00 */ int bWait;
+    /* 0x04 */ pCallback_func* pCallback;
+    /* 0x08 */ u8* pTarget;
+    /* 0x0C */ u32 nSize;
+    /* 0x10 */ u32 nOffset;
+} RomCopyState; // size = 0x14
 
-// size: 0x30
-typedef struct __anon_0x4D1DA {
-    int bWait; // 0x0
-    int bDone; // 0x4
-    int nResult; // 0x8
-    u8* anData; // 0xC
-    int (*pCallback)(); // 0x10
-    int iCache; // 0x14
-    int iBlock; // 0x18
-    int nOffset; // 0x1C
-    u32 nOffset0; // 0x20
-    u32 nOffset1; // 0x24
-    u32 nSize; // 0x28
-    u32 nSizeRead; // 0x2C
-} __anon_0x4D1DA;
+typedef struct RomLoadState {
+    /* 0x00 */ int bWait;
+    /* 0x04 */ int bDone;
+    /* 0x08 */ int nResult;
+    /* 0x0C */ u8* anData;
+    /* 0x10 */ pCallback_func* pCallback;
+    /* 0x14 */ s32 iCache;
```

Comment:
```suggestion
    /* 0x14 */ int iCache;
```

I know we haven't settled in s32 vs int but it would be nice to be consistent in this struct at least